### PR TITLE
Update the handling of joins in DFA to use correct tuple indexes

### PR DESCRIPTION
**Pull request by bgogul**
This also adds a couple of tests with claims on components of a tuple.

### CODE REVIEW

File: java/arcs/core/analysis/InformationFlow.kt [hunk 1]
```diff
@@ -387,6 +385,14 @@ class InformationFlow private constructor(
     )
 
     companion object {
+
+        val tupleIndexNames = listOf("first", "second", "third", "fourth", "fifth")
+
```
Comment (csilvestrini) [inline java/arcs/core/analysis/InformationFlow.kt:390]:
```
val maxTupleSize = tupleIndexNames.size
```

Then use that below?

Comment (bgogul) [reply java/arcs/core/analysis/InformationFlow.kt:390]: Good idea. Done!

File: javatests/arcs/core/analysis/testdata/fail_join_tuple_components.arcs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  claim address.street is sensitive
+
+particle ReadJoinedData
+  details: reads [(
+    &Person {name: Text, age: Number, education: Text},
+    &Address {name: Text, street: Text, city: Text}
+  )]
+  sensitiveInfo: writes Info {info: Text}
+  publicInfo: writes Info {info: Text}
+  claim sensitiveInfo.info derives from details.first.age
```
Comment (piotrswigon) [inline javatests/arcs/core/analysis/testdata/fail_join_tuple_components.arcs:19]: Nice!

**Pull request merged**